plugins/dell-dock/fu-plugin-dell-dock.c

```diff
@@ -129,11 +129,22 @@ fu_plugin_backend_device_added (FuPlugin *plugin,
 	/* GR controller internal USB HUB */
 	if ((guint) fu_usb_device_get_vid (FU_USB_DEVICE (device)) == GR_USB_VID &&
 	    (guint) fu_usb_device_get_pid (FU_USB_DEVICE (device)) == GR_USB_PID) {
+		GPtrArray *devices = fu_plugin_get_devices(plugin);
```

Comment:
this feels too racy to be calling it within `backend_device_added`.  It means the order of events will matter now on which devices have probed first (GR or Hub2).  Can you run it somewhere else?